From 2b4b74b482dda418f60b728b9e5447698b32dda4 Mon Sep 17 00:00:00 2001
From: Svet Ganov <svetoslavganov@google.com>
Date: Thu, 15 Mar 2018 10:45:56 -0700
Subject: [PATCH 159/306] Don't throw but wtf to help chase mismatched
 start/finish op calls

Bug: 74681706

Test: manual

Change-Id: I4661d46f953ad6dc6f11eccc5b3824bf51899374
(cherry picked from commit 31d83ae577c47fff132d4205f1941cd7af13f020)
---
 services/core/java/com/android/server/AppOpsService.java | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/AppOpsService.java b/services/core/java/com/android/server/AppOpsService.java
index ecf80ba7368..9756d17f662 100644
--- a/services/core/java/com/android/server/AppOpsService.java
+++ b/services/core/java/com/android/server/AppOpsService.java
@@ -1346,8 +1346,9 @@ public class AppOpsService extends IAppOpsService.Stub {
                 return;
             }
             if (!client.mStartedOps.remove(op)) {
-                throw new IllegalStateException("Operation not started: uid" + op.uid
+                Slog.wtf(TAG, "Operation not started: uid" + op.uid
                         + " pkg=" + op.packageName + " op=" + op.op);
+                return;
             }
             finishOperationLocked(op, /*finishNested*/ false);
             if (op.nesting <= 0) {
-- 
2.17.1

